### PR TITLE
Refactor renaming master to main

### LINKED ISSUE #327: default branch is 'main' not 'master' now
The new default branch name in GitHub is `main` not `master`. This affects our new repo creation. See attached screenshot with description of modifying local branch name before pushing. 

Screen Shot 2020-11-11 at 5 30 41 PM

I've already got fixes for new repo creation in a local branch as part of my github authentication changes, but wanted to make an explicit note because it may create some confusion with new vs existing repos.

### LINKED ISSUE #363: check for main branch before trying to change
As noted in [this comment](https://github.com/earthlab/abc-classroom/pull/335#discussion_r533625441) on PR #335, we should check whether the `master` branch exists before trying to change it. Suggested fix is:

`git show-ref --quiet --verify refs/heads/master` 

which returns an error code of 1 if the branch does not exist.

### LINKED ISSUE #370: Add git configuration to CI setup
Currently Actions are failing because git is not configured with a username and email. i.e., we need to do:

```
git config --global user.name "John Doe"
git config --global user.email johndoe@example.com
```

### PR DESCRIPTION
**Pull request by kcranston**
This refactors the master-to-main method, including tests for some new things I learned about git (such as, a branch does not exist until there are commits associated with that branch). It also adds a test file for github.py with tests that cover the relevant methods.

abc-classroom no longer prints out the message about changing the name of the branch until it actually does so, and will not try and change the branch name in a repo that already has a `main` branch. 

closes #363 
closes #327
closes #370

### CODE REVIEW

**Comment by lwasser** [comment]
oh gosh - i wonder if CI is out of credits for earth lab too? it's not running

File: abcclassroom/github.py [hunk 1]
```diff
@@ -307,25 +283,43 @@ def init_and_commit(directory, custom_message=False):
...
 
-def _master_branch_to_main(directory):
+def _master_branch_to_main(dir):
     """Change the name of the master branch to main
 
     Changes the name of the master branch to main for the repo in the
     given directory. Since we create the repo on github first, which now sets
     the default branch to 'main', we need the local repo to match
-    in order to be able to push with error later.
+    in order to be able to push without error later.
```
Comment (lwasser) [inline abcclassroom/github.py:292]:
```suggestion
    in order to be able to push without errors later.
```

File: .github/workflows/run-tests-mac-win-linux.yml [hunk 2]
```diff
@@ -31,6 +31,13 @@ jobs:
         python -m pip install --upgrade pip
         pip install -r dev-requirements.txt
         pip install -e .
+    - name: setup git config
+      run: |
+      # setup the git username and email, needed to run tests in test_github
```
Comment (lwasser) [inline .github/workflows/run-tests-mac-win-linux.yml:36]:
```suggestion
      # Setup the git username and email, needed to run tests in test_github
```

**Comment by lwasser** [comment]
ok @kcranston this was failing due to the CI yaml file not being indented correctly.  it l ooks like a few  edits w ere made to i t that threw off the YAML list indentation. it's fixed now so i t can atleast run. You can turn on these actions in your fork to test as well i think altho it may need a secret i can't remember - maybe for  codecov.

**Comment by kcranston** [comment]
Woo-hoo! Finally got this building. Thanks for the help, @lwasser This is ready to merge now.

**Comment by kcranston** [comment]
Building successfully! Thanks for the help, @lwasser This can be merged now.

**Comment by lwasser** [comment]
ok great! merging!

**Pull request merged**